admin-dev/themes/new-theme/js/pages/product/components/combination-list/bulk-delete-handler.ts

```diff
@@ -69,43 +69,80 @@ export default class BulkDeleteHandler {
     }
 
     bulkDeleteBtn.addEventListener('click', () => {
-      try {
-        const selectedCombinationsCount = this.bulkChoicesSelector.getSelectedCheckboxes().length;
-        const confirmLabel = bulkDeleteBtn.dataset.modalConfirmLabel
-          ?.replace(/%combinations_number%/, String(selectedCombinationsCount));
-
-        const modal = new ConfirmModal(
-          {
-            id: 'modal-confirm-delete-combinations',
-            confirmTitle: bulkDeleteBtn.innerHTML,
-            confirmMessage: bulkDeleteBtn.dataset.modalMessage,
-            confirmButtonLabel: confirmLabel,
-            closeButtonLabel: bulkDeleteBtn.dataset.modalCancelLabel,
-            closable: true,
-          },
-          async () => {
-            const response = await this.bulkDelete();
-            $.growl({message: response.message});
-            this.eventEmitter.emit(CombinationEvents.refreshCombinationList);
-          },
-        );
-        modal.show();
-      } catch (error) {
-        const errorMessage = error.response?.JSON ?? error;
-        $.growl.error({message: errorMessage});
-      }
+      this.bulkChoicesSelector.getSelectedIds().then((selectedCombinationIds) => {
+        try {
+          const selectedCombinationsCount = selectedCombinationIds.length;
+          const confirmLabel = bulkDeleteBtn.dataset.modalConfirmLabel
+            ?.replace(/%combinations_number%/, String(selectedCombinationsCount));
+
+          const modal = new ConfirmModal(
+            {
+              id: 'modal-confirm-delete-combinations',
+              confirmTitle: bulkDeleteBtn.innerHTML,
+              confirmMessage: bulkDeleteBtn.dataset.modalMessage,
+              confirmButtonLabel: confirmLabel,
+              closeButtonLabel: bulkDeleteBtn.dataset.modalCancelLabel,
+              closable: true,
+            },
+            async () => {
+              await this.bulkDelete(selectedCombinationIds);
+              //@todo: hardcoded success. Pass translated message to modal data attribute?
```

Comment:
It's ok for now, this will be replaced by the progress modal anyway

Reply:
are you sure? its confirmation modal. Will new confirmation modal also be handled in new progress modal?